### DISCUSSION TITLE
iOS SDK doesn't support priority fee for Solana

### DISCUSSION THREAD

**Discussion by ditzdragos** [trustwallet/wallet-core#3773] [Q&A]
**Describe the bug**
In the latest Trust Wallet SDK version (4.0.32) for iOS, the support for priority fee is missing in the interface for Solana transactions, particularly in the SigningInput component.

**To Reproduce**

1. Initialize Trust Wallet SDK version 4.0.32 on iOS.
2. Attempt to implement transaction signing for Solana blockchain.
3. Note that there is no option or interface provided for setting a priority fee.

**Expected behavior**
I expected the Trust Wallet SDK to include support for setting a priority fee for Solana transactions, similar to the Android version.

**Additional context**
It's worth noting that the Android version of Trust Wallet SDK does support priority fee for Solana transactions. This inconsistency between platforms needs to be addressed to ensure parity in functionality across different operating systems.

**Comment by satoshiotomakan** [comment]
Hi @ditzdragos, could you please try to clean the cache after updating to 4.0.32?
We've tested the functionality, and `SigningInput::priority_fee_price` and `SigningInput::priority_fee_limit` are used by other teams already.
Please take a look at the Rust example on the recently added fields:
https://github.com/trustwallet/wallet-core/blob/35378aa5f420cf70319cf58cbadcc8ccc52ce8aa/rust/tw_any_coin/tests/chains/solana/solana_sign.rs#L141

**Comment by ditzdragos** [comment]
@satoshiotomakan We tried various methods for refreshing the state of dependencies and the fields are still missing.

**Reply by satoshiotomakan** [reply]
@ditzdragos, could you please try to upgrade to 4.0.33 release?
Please also make sure if you use the recently added fields properly, e.g as follows:
```swift
        let transferMessage = SolanaTransfer.with {
            $0.recipient = "EN2sCsJ1WDV8UFqsiTXHcUPUxQ4juE71eCknHYYMifkd"
            $0.value = 42
        }
        let input = SolanaSigningInput.with {
            $0.transferTransaction = transferMessage
            $0.recentBlockhash = "11111111111111111111111111111111"
            $0.privateKey = privateKeyData
            $0.priorityFeeLimit = SolanaPriorityFeeLimit.with {
                $0.limit = 2000
            }
            $0.priorityFeePrice = SolanaPriorityFeePrice.with {
                $0.price = 2
            }
        }

        let output: SolanaSigningOutput = AnySigner.sign(input: input, coin: .solana)
```

**Reply by satoshiotomakan** [reply]
Please also clarify if you use SPM or Cocoapods package?

**Reply by ditzdragos** [reply]
We use SPM.
4.0.33 doesn't seem to work for us
![image](https://github.com/trustwallet/wallet-core/assets/11709207/939908c4-0f5c-4a84-be2c-dfd8751d3bb3)
![image](https://github.com/trustwallet/wallet-core/assets/11709207/07742bdf-8521-4341-a063-d7b02b4ba5e2)

**Comment by luciansvnc** [comment]
Actually I am having the same issues on iOS. The SolanaSigningInput struct doesn't have priority fee properties.

**Comment by satoshiotomakan** [comment]
The problem should be solved at https://github.com/trustwallet/wallet-core/pull/3775
